### ISSUE TITLE
Why does compute query loss in training_step?

### ISSUE THREAD

**Issue by minocrafft** [mikwieczorek/centroids-reid#33]
First of all, thank you for your research with a good idea.

While analyzing the code of the ctl model, I left an issue because I was curious.

[training_step code](https://github.com/mikwieczorek/centroids-reid/blob/a1825b7a92b2a8d5e223708c7c43ab58a46efbcf/train_ctl_model.py#L62)
```python
        # query in training_step
        contrastive_loss_query, _, _ = self.contrastive_loss(
            features, class_labels, mask=isReal
        )
        contrastive_loss_query = (
            contrastive_loss_query * self.hparams.SOLVER.QUERY_CONTRASTIVE_WEIGHT
        )

        class_labels_real = class_labels[isReal]
        features_real = features[isReal]
        center_loss = self.hparams.SOLVER.CENTER_LOSS_WEIGHT * self.center_loss(
            features_real, class_labels_real
        )
        bn_features = self.bn(features_real)
        cls_score = self.fc_query(bn_features)
        xent_query = self.xent(cls_score, class_labels_real)
        xent_query = xent_query * self.hparams.SOLVER.QUERY_XENT_WEIGHT
```

From that link, the query data is calculated with contrastive loss, center loss, and xent loss, and then added to the total_loss at the bottom.
As far as I know, query and gallery dataset is used for validation.

Isn't the data loaded into the batch of training_step the data of the train dataset?
Or does *_loss_query have any other meaning?
Did I miss anything?

If anyone has the same problem as me, I would appreciate it if you could reply.
